### ISSUE TITLE
Node Dashboard Empty values after suspended state.

### ISSUE THREAD

**Issue by JavierRefuerzo** [UniversalDevicesInc/UD-Mobile-iOS#13]
The Node Dashboard screen will have no values after a suspended app state. There is no refresh after app connection

**Comment by BitArtOnLine** [comment]
I'm seeing this differently. When the app is first opened, sync'd, I never see any statuses. If I then suspend the app (switch-away or close to phone home), then upon opening app again statuses immediately populate.

**Comment by JavierRefuerzo** [comment]
This issue specifically is related to the screen showing a single node (Node Dashboard).  When opened from the list of nodes (Home Screen).   If opened from the Home Screen values are populated, if app is then suspended, then user returns to the app the Node Dashboard remains empty.

Are you referring to the Home Screen or Node Dashboard?

**Comment by BitArtOnLine** [comment]
It applies to all nodes (Devices and Scenes) after app is initially opened.



After suspending the app and reopening, statuses appear (values populate quickly - nice operation).

**Comment by BitArtOnLine** [comment]
App install;
Sync;
Open Devices and Scenes - No statuses;
Navigate/open folders - no statuses;
Waited for 5 minutes (timed) - no statuses;
Suspend app;
Re-open - statuses populate

Christopher Green

**Comment by JavierRefuerzo** [comment]
Thanks Christopher, 
If I remember correctly you have multiple iOS devices. Did this happen with every device?  If it only happens on one device please let me know the model and iOS version and I'll try to replicate on the emulator.  If it is not consistent and you see this again please pull down your notifications. Notifications are used to show connection status, there should be one notification for each ISY.   Note the connection status.

I'm guessing there may the socket may be waiting on a connection or there was a connection issue. Issues should trigger a drop down notification.  I'll have to check the timeout for the initial connection, after connected the app will attempt to reconnect if the heartbeat is not received or the socket is disconnected.

**Comment by BitArtOnLine** [comment]
Happened initially on iPhone and then I took it step at a time on iPad since was looking at the local net DB issue.

I get a notification that sync occurred successfully (still have one NS that causes an error when it’s parsed). Can’t find any place in app that shows status notifications on demand. 

Remember, I have a lot of nodes in the ISY. So far, due to that, I believe I’ve seen obscure occurrences.

Christopher Green

**Comment by JavierRefuerzo** [comment]
Hi Christopher,
I forgot about the parsing issue,  I'll take a look at that files soon.  
As for the notification I am referring to your system notifications, i.e push notifications.  The app sets one for each ISY while they are connected.  They **_should_** drop down automatically if there is a connection issue, if there is no issue the status is still visible when you pull down your notifications bar.

**Comment by JavierRefuerzo** [comment]
@BitArtOnLine I think your issue was caused by #14 . If you see this again please open a new issue.

Fixed Node Dashboard Empty values after suspended state by removing whole query observer on complex query. This caused DB access to become blocked by observer.

**Comment by BitArtOnLine** [comment]
Yes from your fix description it sounds right. Interesting language iPhone has, very complete but I can sense it is prickly.

Christopher
